svg.panzoom.js

```diff
@@ -129,16 +126,47 @@ SVG.extend(SVG.Doc, SVG.Nested, {
       lastP = currentP
     }
 
-    this.on('wheel', zoom)
+    this.on('wheel', wheelZoom)
     this.on('touchstart', pinchZoomStart, this, {passive:false})
     this.on('mousedown', panStart, this, {passive:false})
-    this.on('touchstart', panStart, this, {passive:false})
 
     return this
 
-  }
+  },
 
+  zoom: function(level, point) {
+    var style = window.getComputedStyle(this.node)
+      , width = parseFloat(style.getPropertyValue('width'))
+      , height = parseFloat(style.getPropertyValue('height'))
+      , v = this.viewbox()
+      , zoomX = width / v.width
+      , zoomY = height / v.height
+      , zoom = Math.min(zoomX, zoomY)
+
+    if(level == null) {
+      return zoom
+    }
+
+    var zoomAmount = (zoom / level)
+
+    point = point || new SVG.Point(width/2 / zoomX + v.x, height/2 / zoomY + v.y)
+
+    var box = new SVG.Box(v)
+      .transform(new SVG.Matrix()
+        .scale(zoomAmount, point.x, point.y)
+      )
+
+    if(this.fire('zoom', {box: box, focus: point}).event().defaultPrevented)
+      return this
+
+    return this.viewbox(box)
```

Comment:
That does not cover `wheelZoom` and programmable zoom. But I think we can get away with only having one `zoom` event. I'll need to play with it a little